internal/pkg/skuba/addons/kured_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package addons
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	img "github.com/SUSE/skuba/pkg/skuba"
+)
+
+func TestGetKuredImage(t *testing.T) {
+	tests := []struct {
+		name     string
+		imageTag string
+		want     string
+	}{
+		{
+			name:     "get kured image without revision",
+			imageTag: "1.2.0",
+			want:     img.ImageRepository + "/kured:1.2.0",
+		},
+		{
+			name:     "get kured image with revision",
+			imageTag: "1.2.0-rev4",
+			want:     img.ImageRepository + "/kured:1.2.0-rev4",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetKuredImage(tt.imageTag); got != tt.want {
+				t.Errorf("GetKuredImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_KuredImage(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	tests := []test{}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tests = append(
+			tests,
+			test{
+				name: "render kured image URL when cluster version is " + ver.String(),
+				renderContext: renderContext{
+					config: AddonConfiguration{
+						ClusterVersion: ver,
+						ControlPlane:   "",
+						ClusterName:    "",
+					},
+				},
+				want: img.ImageRepository + "/kured:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
```

Comment:
I don't see we have this implementation in code to check image tag format. If don't have this code, why we need to have this unit test? 

I would suggest to drop this test case, because it is already covered in `GetKuredImage`. How do you think?

Reply:
hmm, I think we still need to verify the public interface even the implementation we already cover in unit tests. Revoking my request :)